R Guide/R Kaggle Guide (Titanic).Rmd

```diff
@@ -0,0 +1,96 @@
+
+The `$` let's us select specific variables in a dataframe.
+
+```{R}
+titanic_data$Survived <- as.factor(titanic_data$Survived)  
+titanic_data$Pclass <- as.factor(titanic_data$Pclass)
+titanic_data$Sex <- as.factor(titanic_data$Sex)
+titanic_data$Embarked <- as.factor(titanic_data$Embarked)
+```
+
+We can observe the first `n` entries of our dataframe by using the `head()` function, likewise we to observe the last `n` entires we can use `tail()`. If there are too many variables, the output will omit them to save space.
+
+```{R}
+head(titanic_data, 5)
+```
+After a quick look, let's get a summary of our data.
+```{R}
+summary(titanic_data)
+```
+The `NA's` in some columns indicate the number of missing values. One could either remove the rows with missing values, or try to fill in the data based on surrounding data. Since our dataset is fairly small, the latter is preferred. This is called **imputation**.
+
+Let's get a closer look at who these people with missing embarked locations are. We can use the `filter()` function to select rows that satisfy a certain criteria. Note that we do not have to use $ to indicate that `Embarked` is from `titanic_data` because it's inferred when we put what data we're looking at as the first parameter in `filter`.
+
+**NOTE:** `NA == NA` will return `NA`. While this may be confusing think of it this way. 
+```{R}
+alice.age <- NA  # We don't know Alice's age
+bob.age  <- NA  # We don't know Bob's age
+alice.age == bob.age  # Are Alice and Bob the same age? We don't know!
+```
+
+That's why we use `is.na` to test for missing values instead.  
+
+**NOTE:** The code below might start to look a little convoluted. We'll soon look at some syntactic sugar to make everything easier to read.
+
+```{R}
+filter(titanic_data, is.na(Embarked))[c('Name', 'Fare', 'Ticket', 'Cabin')]
+```
+It seems the passenger's had the same ticket, hence the identical fare. Let's visualize how much a passenger paid and their class based off the location and they embarked. We add $80 as the dashed red line to make a comparison.
+```{R}
+ggplot(filter(titanic_data, !is.na(Embarked)), 
+       aes(x = Embarked, y = Fare, fill = Pclass)) +
+  geom_boxplot() +
+  scale_y_continuous() +
+  labs(title = "Ticket Price from Embark Location",
+       y = "Fare [$]") +
+  geom_hline(aes(yintercept = 80),
+             colour = "red", linetype = "dashed", lwd = 1)
+```
+
+The red line is aligned with the median of the fare paid in location C. Thus we will fill in the missing embarked locations with C. 
+
+```{R}
+titanic_data$Embarked[is.na(titanic_data$Embarked)] <- 'C'
+```
+
+## INCOMPLETE SECTION
+
+Another method of imputation is through prediction. It would be naive to use simple methods such as mean because we have other data that hint towards the age of a passenger. We can make a model to estimate the age from the other information we have.
```

Comment:
Please include a section of missing data mechanisms. 

more information can be found in `Elements of Statistical Learning` in the missing data section. 

basically that missing data in itself can be predictive and we can always include `is.na(feature)` as a new indicator variable feature.